scm-ui/e2e-tests/cypress/integration/anonymous_mode_full_spec.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020-present Cloudogu GmbH and Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { hri } from "human-readable-ids";
+
+describe("Anonymous Mode Full", () => {
+  beforeEach(() => {
+    cy.restSetAnonymousMode("FULL");
+  });
+
+  it("should show login button on anonymous route", () => {
+    // Arrange
+    cy.restSetUserPermissions("_anonymous", ["repository:read,pull:*"]);
+    // Act
+    cy.visit("/repos/");
+    // Assert
+    cy.byTestId("repository-overview-filter");
+    cy.byTestId("primary-navigation-login");
```

Comment:
This new test format is very explicit. It feels kinda difficult to read and not reuseable. We should atleast add another layer with most common functions. Maybe like `setRepoReadPermission(username)`?